infra/base-images/base-builder/compile_honggfuzz

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -eu
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+echo -n "Compiling honggfuzz to $LIB_FUZZING_ENGINE ..."
+
+pushd $SRC/honggfuzz > /dev/null
+make clean
+# Compile honggfuzz with gcc, as clang will be source of problems
+# for libhfuzz/libhfuzz.a
+CC=gcc CFLAGS= make
+# It'll be added by CC/CXX directly during linking, so its defined here
+# as a duplicate to satisfy building infrastructure
+cp libhfuzz/libhfuzz.a $LIB_FUZZING_ENGINE
+cp honggfuzz hfuzz_cc/hfuzz-clang hfuzz_cc/hfuzz-clang++ $OUT/
+popd > /dev/null
+
+# Relevant coverage flags are added by the compiler/linker
+export COVERAGE_FLAGS=
```

Comment:
Nah.. here's what we add

Compiler:
https://github.com/google/honggfuzz/blob/master/hfuzz_cc/hfuzz-cc.c#L131

Linker:
https://github.com/google/honggfuzz/blob/master/hfuzz_cc/hfuzz-cc.c#L202

So it's only coverage-related options plus libhfuzz.a when linking (to avoid the problem that clang addss it's own _cov_trace_pc symbols when compiling with -fsanitize=address|memory|undefined).